examples/example.go

```diff
@@ -1,39 +1,33 @@
 package main
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
-	"github.com/sendgrid/rest"
+	"net/http"
+	"net/url"
 	"os"
+
+	"github.com/sendgrid/rest"
 )
 
 func main() {
-
 	// Build the URL
 	const host = "https://api.sendgrid.com"
 	endpoint := "/v3/api_keys"
-	baseURL := host + endpoint
-
-	// Build the request headers
 	key := os.Getenv("SENDGRID_API_KEY")
-	Headers := make(map[string]string)
-	Headers["Authorization"] = "Bearer " + key
-
-	// GET Collection
-	method := rest.Get
-
-	// Build the query parameters
-	queryParams := make(map[string]string)
-	queryParams["limit"] = "100"
-	queryParams["offset"] = "0"
-
-	// Make the API call
-	request := rest.Request{
-		Method:      method,
-		BaseURL:     baseURL,
-		Headers:     Headers,
-		QueryParams: queryParams,
-	}
+
+	// GET
+	baseURL, _ := url.Parse(host + endpoint)
```

Comment:
Unnecessary parse call anyway as noted below. Using a struct literal is much more idiomatic and clear. Awesome suggestion.